### DISCUSSION TITLE
authorization error when insert data

### DISCUSSION THREAD

**Discussion by wusixuan0** [supabase/supabase#28359] [Questions] [answered]
Hello everyone,

I'm encountering an authorization error while trying to seed data using the Python SDK for Supabase. I'm hoping someone might have experience with this issue and can offer some guidance.

The error I'm receiving is: Error inserting data: {'code': '42501', 'details': None, 'hint': None, 'message': 'permission denied for schema public'}

Some context:

This seeding script was working correctly before today.

Today, I ran the following commands to reset my schema: DROP SCHEMA public CASCADE; CREATE SCHEMA public;

After that, I ran Django migrations to regenerate the database schema.

Now, when I try to run my seeding script using the Python SDK, I get the above error.

I've already double-checked permissions for my API key and Postgres user permissions, which appear to be correct

Has anyone encountered a similar issue after resetting their schema? Any ideas on what might be causing this and how to resolve it?

Thank you in advance for any help or insights you can provide!

**Comment by GaryAustin1** [comment]
You wiped your default Supabase permissions.
```sql
GRANT USAGE ON SCHEMA myschema TO anon, authenticated, service_role;
GRANT ALL ON ALL TABLES IN SCHEMA myschema TO anon, authenticated, service_role;
GRANT ALL ON ALL ROUTINES IN SCHEMA myschema TO anon, authenticated, service_role;
GRANT ALL ON ALL SEQUENCES IN SCHEMA myschema TO anon, authenticated, service_role;
ALTER DEFAULT PRIVILEGES FOR ROLE postgres IN SCHEMA myschema GRANT ALL ON TABLES TO anon, authenticated, service_role;
ALTER DEFAULT PRIVILEGES FOR ROLE postgres IN SCHEMA myschema GRANT ALL ON ROUTINES TO anon, authenticated, service_role;
ALTER DEFAULT PRIVILEGES FOR ROLE postgres IN SCHEMA myschema GRANT ALL ON SEQUENCES TO anon, authenticated, service_role;
```
With myschema set to public will fix it.
https://supabase.com/docs/guides/api/using-custom-schemas

**Reply by GaryAustin1** [reply, marked as answer]
Edit:  I just looked at the Prisma troubleshooting guide and it shows this for the public schema.   Note the grant to postgres... not sure if that is needed or not depending on the main role being used.
```sql
grant usage on schema public to postgres, anon, authenticated, service_role;

grant all privileges on all tables in schema public to postgres, anon, authenticated, service_role;
grant all privileges on all functions in schema public to postgres, anon, authenticated, service_role;
grant all privileges on all sequences in schema public to postgres, anon, authenticated, service_role;

alter default privileges in schema public grant all on tables to postgres, anon, authenticated, service_role;
alter default privileges in schema public grant all on functions to postgres, anon, authenticated, service_role;
alter default privileges in schema public grant all on sequences to postgres, anon, authenticated, service_role;
```

**Reply by wusixuan0** [reply]
Thank you so much!! This worked!
